rasterio/plot.py

```diff
@@ -130,17 +134,30 @@ def show(source, with_bounds=True, contour=False, contour_label_kws=None,
     return ax
 
 
-def plotting_extent(source):
+def plotting_extent(source, transform=None):
     """Returns an extent in the format needed
      for matplotlib's imshow (left, right, bottom, top)
      instead of rasterio's bounds (left, bottom, top, right)
 
     Parameters
     ----------
-    source : raster dataset
+    source : raster dataset or array
+    transform: Affine, required if source is array
     """
-    extent = (source.bounds.left, source.bounds.right,
-              source.bounds.bottom, source.bounds.top)
+    try:
+        extent = (source.bounds.left, source.bounds.right,
+                  source.bounds.bottom, source.bounds.top)
+    except AttributeError:
+        # no bounds attribute, try array
+        if transform is None:
+            raise ValueError(
+                "transform is required if source is an array")
+        transform = guard_transform(transform)
+        rows, cols = source.shape[0:2]
```

Comment:
at the moment, this function assumes that the array has already been converted to image order (https://github.com/mapbox/rasterio/blob/80a5733acebf1a29bebc8846709443ecebc9f1e6/rasterio/plot.py#L102). This may not be the best approach but it seemed the cleanest given that the input could also be a 2D array (1 band)